app-typescript/components/Checkbox.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+
+interface IProps {
+    value: any;
```

Comment:
Why is value `any` instead of boolean?

Reply:
Checkbox values should not be only `true` or `false`. Sometimes you need to assign certain value to checkbox. This is also supported by native checkbox: https://www.w3schools.com/tags/att_input_type_checkbox.asp

Reply:
Also here we can have same discussion should this be object, boolean or string. It can be everything.

Reply:
The checkbox will be either checked or not, that's why I suggest having a boolean. What I assume you would do if an object is passed is check if it's `null` and convert it to boolean that way. I don't see a reason to put it inside component logic since it's trivial to do from outside of the component:
```
<Checkbox value={myObject != null} onChange={...} />
```

Reply:
No, sometimes on `value` you want to assign `test1` not just `true`. Check link I send it to you, as you can see, basic functionality

Reply:
If you pass `'test1'`, how will you decide if the checkbox is supposed to be checked or not? By converting it to boolean while checking if it's a string and not empty? The same can easily be done outside the component identically to the object example:

```
<Checkbox value={!isEmtpy(str)} onChange={...} />
```

You are also losing type safety by allowing objects. What if an object is empty? What if it's an array, Set, Map? It's just easier to have the consumer of the component handle it and pass the boolean value.

Reply:
What you will get as response on onChange? `True` or `False`, not actual value. As I sad, check what is native behaviour of checkboxes.

Reply:
If I need object, I will use object. Check other frameworks, all are allowing objects and setting custom values for them. What are we? Smartest people on planet and we want to make life of developers difficult? Idea of `superdesk-ui-framework` is to make use of components easier, not to make it to complicate.

Reply:
If you pass a boolean in as a value, you get a boolean out on `onChange` too.

Let's save time on reviews: if you can, you review the components yourself. If you want me to take a look, add me as a reviewer. I'll post suggestions and you take away from there and decide which suggestions you want to implement :wink:

Reply:
Yes, and you want to have string as output of `onChange` so you can assign it to value. Simple as that. No, here we are discussing about stupid things, trivial, something that is already implemented as native functionality of same component. Let's not lose time on stupid discussions and trivial things. Suggestions and reviews are important, but not like this. Let's focus on functionalities on specific component, not on types and naming (I'm not saying it's not important, but it is not big deal if something is called `value` or `model`, also agreeing to have something consistant). But discussions like this are exhausting and it's just case how developer is seeing something. And at the end you are trying to find arguments just to justify your arguments. We are not kids, we have kids.

Reply:
I don't see why we would need a value property here. what about:
```
interface IProps {
    checked: boolean;
    onChange: (checked: boolean) => void;
}

Reply:
That's the same as what I was suggesting, except that I was calling the `on/off` state `value: boolean` instead of `checked: boolean`. I don't mind which name is used.

Reply:
Ok, so not handling any value inside `Checkbox` component, just visually adding checkbox sign. This is ok for `Checkbox` but for `Radio` we will need to support value. Because of consistency I tough to implement same thing in `Checkboxes` because they are pretty similar components. So for the end user not to be confused about implementation. One component is returning boolean and other one is returning value. To make it similar as much as possible.

Reply:
I like things consistent, but explicit is even better :wink:

Reply:
![image](https://user-images.githubusercontent.com/6507688/69860975-9bb65180-1297-11ea-9d1b-fd8267ca32cf.png)

Reply:
https://gph.is/10qGOQg